libibverbs/device.c

```diff
@@ -262,6 +267,8 @@ struct ibv_context *__ibv_open_device(struct ibv_device *device)
 	context->cmd_fd = cmd_fd;
 	pthread_mutex_init(&context->mutex, NULL);
 
+	ibverbs_device_hold(device);
+
```

Comment:
Why do we need a solution for everything that touches sysfs ? same TS on the same sysfs device name entry should guaranteed same device .
 In addition, extra checking before attempting to open looks redundant, as just after checking it might be dropped and there is no atomicy with kernel here. However, doing the check after opening the cmf_fd will make sure that if later the device will be dropped the alloc context command will fail and the application will get an error as we already agreed.